kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ResolutionResult.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.resolver;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaClusterRef;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.operator.ResourcesUtil;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static java.util.Comparator.comparing;
+
+/**
+ * The result of a deep resolution of the dependencies of a single KafkaProxy. Contains all
+ * Filters, KafkaProxyIngresses and KafkaClusterRefs that were successfully resolved. Contains
+ * a list of unresolved references per VirtualKafkaCluster.
```

Comment:
It is a map of cluster recon results.

Reply:
I was more trying to say that ResolutionResult describes which dependencies are unresolved per virtual cluster, have reworded.

Reply:
added javadoc to it's methods as well